hawkular-command-gateway-itest/src/test/java/org/hawkular/cmdgw/ws/test/TestWebSocketClient.java

```diff
@@ -46,24 +52,112 @@
 import org.hawkular.bus.common.BasicMessageWithExtraData;
 import org.hawkular.cmdgw.api.ApiDeserializer;
 import org.hawkular.cmdgw.api.WelcomeResponse;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ActualEvent.ActualClose;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ActualEvent.ActualFailure;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ActualEvent.ActualMessage;
+import org.hawkular.cmdgw.ws.test.TestWebSocketClient.ExpectedEvent.ExpectedMessage;
 import org.testng.Assert;
 
+import com.squareup.okhttp.MediaType;
 import com.squareup.okhttp.OkHttpClient;
 import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
 import com.squareup.okhttp.Response;
+import com.squareup.okhttp.ResponseBody;
 import com.squareup.okhttp.ws.WebSocket;
-import com.squareup.okhttp.ws.WebSocket.PayloadType;
 import com.squareup.okhttp.ws.WebSocketCall;
 import com.squareup.okhttp.ws.WebSocketListener;
 
 import okio.Buffer;
 import okio.BufferedSource;
+
 /**
  * A client for testing WebSocket conversations. A typical usage starts by {@link TestWebSocketClient#builder()}.
  *
  * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
  */
 public class TestWebSocketClient implements Closeable {
+
+    public abstract static class ActualEvent {
+        public static class ActualOpen extends ActualEvent {
+            public ActualOpen(TestListener testListener, int index, WebSocket webSocket, Response response) {
+                super(testListener, index);
+                this.webSocket = webSocket;
+                this.response = response;
+            }
+            private final WebSocket webSocket;
```

Comment:
That is a mistake. Eclipse is not able to place the generated constructors where they belong, there are just options like current cursor position, top, bottom, which mostly wrong and here, I apparently forgot to reorder. Will fix.